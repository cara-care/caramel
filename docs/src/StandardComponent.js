import React from 'react';
import './assets/css/standardcomponent.css';

const StandardComponent = () => {
  return (
    <div>
      <h2>Accordion</h2>
      <div className="imageTableContainer">
        <div>
          <img style={{maxWidth: "300px", margin: "24px"}} src={require("./assets/images/Accordion.gif")} />
        </div>
        <div className="divTable">
          <div className="divTableBody">
            <div className="divTableRow divTableHeading">
              <div className="divTableCell">Property</div>
              <div className="divTableCell">Type</div>
              <div className="divTableCell">Default</div>
              <div className="divTableCell">Description</div>
            </div>
            <div className="divTableRow">
              <div className="divTableCell">
                <pre className="codeArea">
                  {`
list: {
  image ?: ImageURISource;
  name: string;
  description: string;
  iconStyle?: ImageStyle;
  titleStyle?: TextStyle;
  descriptionStyle?: TextStyle;
  arrowStyle?: ImageStyle;
  separatorStyle?: ViewStyle;
  rowContainerStyle?: ViewStyle;
  onOpen?: () => void;
  onClose?: () => void;
}[]
                  `}
                </pre>
              </div>
              <div className="divTableCell">Array</div>
              <div className="divTableCell">-</div>
              <div className="divTableCell">The list array to render the AccordionRow components. You can add events and
            styles per row.</div>
            </div>
            <div className="divTableRow">
              <div className="divTableCell">
                style
          </div>
              <div className="divTableCell">ViewStyle |Â undefined</div>
              <div className="divTableCell">-</div>
              <div className="divTableCell">Style for the FlatList parent</div>
            </div>
            <div className="divTableRow">
              <div className="divTableCell">
                disableAutoClose
          </div>
              <div className="divTableCell">boolean | undefined</div>
              <div className="divTableCell">false</div>
              <div className="divTableCell">Disables the rows closing themselves automatically when another row is
            opened within the same Accordion.</div>
            </div>
            <div className="divTableRow">
              <div className="divTableCell">
                animate
          </div>
              <div className="divTableCell">boolean | undefined</div>
              <div className="divTableCell">false</div>
              <div className="divTableCell">Animates the opening and closing events of the rows.</div>
            </div>
          </div>
        </div>
      </div>
      <h4>Example</h4>
      <pre className="codeArea">
        {
`<Accordion
  list={[
      {
        image: require('./caramel.png'),
        name: 'Accordion1',
        description: 'Lorem ipsum dolor sit amet.',
      },
      {
        image: require('./caramel.png'),
        name: 'Accordion2',
        description: 'Lorem ipsum dolor sit amet.',
      },
    ]}
  animate />`
  }
  </pre>
    </div>
  );
}

export default StandardComponent;
