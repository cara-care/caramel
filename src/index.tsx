export {default as Accordion} from './components/Accordion';
export {default as AccordionRow} from './organisms/AccordionRow';
export {default as AnimatedProgressBar} from './components/AnimatedProgressBar';
export {default as BottomActionSheet} from './components/BottomActionSheet';
export {default as ColorText} from './components/ColorText';
export {default as Container} from './components/Container';
export {default as CoverScrollView} from './components/CoverScrollView';
export {default as IconButton} from './components/IconButton';
export {default as Slider} from './components/Slider';
export {default as Text} from './components/Text';
export {default as ThermoScore} from './components/ThermoScore';
export {
  default as TouchableBreath,
  TouchableBreathProps,
} from './components/TouchableBreath';
export {default as theme} from './utils/Theme';
