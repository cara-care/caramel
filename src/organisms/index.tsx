import BottomSheetAndroid from './BottomSheetAndroid';
import AccordionRow from './AccordionRow';
import CoverImage from './CoverImage';

export {BottomSheetAndroid, AccordionRow, CoverImage};
