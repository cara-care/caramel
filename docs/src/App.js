import React from 'react';
import StandardComponent from './StandardComponent';

const App = () => {
  return (
    <div>
      <StandardComponent />
    </div>
  );
}

export default App;
