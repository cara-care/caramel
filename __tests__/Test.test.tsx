import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Text} from '../src/';

describe('<Container />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Text>Text</Text>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
