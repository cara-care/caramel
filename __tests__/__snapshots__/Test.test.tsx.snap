// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> renders correctly 1`] = `
<Text
  style={
    Array [
      Array [
        Object {
          "fontSize": 16,
          "letterSpacing": -0.28,
          "lineHeight": 24,
        },
        Object {
          "color": "#a9aebe",
          "marginBottom": 0,
        },
      ],
      undefined,
    ]
  }
>
  Text
</Text>
`;
